Makefile

```diff
@@ -0,0 +1,88 @@
+PWD := $(shell pwd)
+
+ifeq ($(RADSAN_BUILD_DIR),)
+	BUILD_DIR := $(PWD)/build
+else
+	BUILD_DIR := $(RADSAN_BUILD_DIR)
+endif
+
+NPROCS := 1
+OS := $(shell uname -s)
+
+ifeq ($(OS),Linux)
+	NPROCS := $(shell grep -c ^processor /proc/cpuinfo)
+endif
+ifeq ($(OS),Darwin)
+	NPROCS := $(shell sysctl -n hw.ncpu)
+endif
+
+CCACHE := $(shell which ccache)
+
+clang: generate
+	cmake --build $(BUILD_DIR) --target clang compiler-rt llvm-symbolizer -j$(NPROCS)
+	echo "\nclang output:\n" && find $(BUILD_DIR)/bin -type l | grep clang;
+
+build-folder:
+	if [ ! -d $(BUILD_DIR) ]; then \
+		mkdir -p $(BUILD_DIR); \
+	fi
+
+submodules:
+	if [ ! -f llvm-project/README.md ]; then \
+		git submodule update --init --recursive; \
+	fi
+
+generate: build-folder submodules
+	if [ ! -f $(BUILD_DIR)/CMakeCache.txt ]; then \
+		cmake -G "$${RADSAN_CMAKE_GENERATOR:-Ninja}" \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DBUILD_SHARED_LIBS=ON \
+		-DCOMPILER_RT_BUILD_SANITIZERS=ON \
+		-DLLVM_ENABLE_PROJECTS="clang;compiler-rt" \
+		-DLLVM_TARGETS_TO_BUILD=Native \
+		-DCMAKE_C_COMPILER_LAUNCHER=$(CCACHE) \
+		-DCMAKE_CXX_COMPILER_LAUNCHER=$(CCACHE) \
+		-DSANITIZER_MIN_OSX_VERSION=$${RADSAN_MIN_OSX_VERSION:-10.15} \
+		-B $(BUILD_DIR) \
+		./llvm-project/llvm; \
+	fi
+
+test-radsan: generate clang
+	cmake --build $(BUILD_DIR) --target TRadsan-arm64-NoInstTest TRadsan-arm64-Test -j$(NPROCS)
+	$(BUILD_DIR)/projects/compiler-rt/lib/radsan/tests/Radsan-arm64-NoInstTest
+	$(BUILD_DIR)/projects/compiler-rt/lib/radsan/tests/Radsan-arm64-Test
+
+check-compiler-rt: generate
+	cmake --build $(BUILD_DIR) --target check-compiler-rt -j$(NPROCS)
+
+docker:
+	docker build -t radsan -f $(PWD)/docker/Dockerfile .
+
+# Does not include check-compiler-rt as it is extremely slow
+test: clang test-radsan
```

Comment:
If `test-radsan` depends on the `clang` target, do we need this `test` target as well? It's a bit difficult for me to understand what the difference is intended to be from the help message below... did you mean to get rid of one of them, maybe?

Reply:
Good catch, a vestigial limb of an earlier approach